File: lei_lds2122_grupo2-main/Frontend/src/app/viewModels/requests/CreateUseRouteVM.ts
import { toJSONString } from "../interfaces/toJSONString"
import { RoutePlaceType } from "../Types/RoutePlaceType"

/**
 * View Model class for Route creatin and verification requests.
 */
export class CreateUseRouteVM implements toJSONString {

    /**
     * Construct's an instance of {@link CreateUseRouteVM}
     * 
     * @param places places used on the route and it's visiting times.
     * @param name route's name
     */
    constructor(
        public places: Array<RoutePlaceType>,
        public name: string
    ) { }

    /**
     * Converts the class into a string representation of a json object 
     * 
     * @returns string representation of the object
     */
    public toJSONString(): string {
        return JSON.stringify(this)
    }
}

File: Backend/Data/ViewModels/Request/CreateUseRouteVM.cs
using Backend.Data.ViewModels.Types;

namespace Backend.Data.ViewModels
{
    /// <summary>
    /// View Model class for Creat and Use Route, this can be used to either created and generated routes reqeusts or to validate a route request
    /// </summary>
    public class CreateUseRouteVM
    {
        public string Name{get; set;}
        public List<RoutePlaceType> Places{get; set;}

        /// <summary>
        /// Class' constructor method
        /// </summary>
        /// <param name="name">Route's name</param>
        /// <param name="places">List of places that the route has</param>
        public CreateUseRouteVM(string name, List<RoutePlaceType> places){
            this.Name = name;
            this.Places = places;
        }
    }
}